win32ss/user/ntuser/winsta.c

```diff
@@ -537,11 +550,64 @@ NtUserCreateWindowStation(
         return NULL;
     }
 
+    /*
+     * If the caller did not provide a window station name, build a new one
+     * based on the logon session identifier for the calling process.
+     */
+    if (!LocalObjectAttributes.ObjectName)
+    {
+        LUID CallerLuid;
+        WCHAR ServiceWinStaName[MAX_PATH];
```

Comment:
MAX_PATH is not a good value here (ideally it wouldn't even be defined in kernel mode). The buffer should probably just be allocated from pool with the right size (i.e. some equivalent of `gustrWindowStationsDir.Length + strlen("\\Service-0x12345678-12345678$")*sizeof(WCHAR) + sizeof(UNICODE_NULL)`)